crates/services/p2p/src/peer_manager.rs

```diff
@@ -132,27 +132,18 @@ impl PeerManager {
     pub fn handle_peer_connected(
         &mut self,
         peer_id: &PeerId,
-        addresses: Vec<Multiaddr>,
         initial_connection: bool,
     ) -> bool {
         if initial_connection {
-            self.handle_initial_connection(peer_id, addresses)
+            self.handle_initial_connection(peer_id)
         } else {
-            let peers = self.get_assigned_peer_table_mut(peer_id);
-            insert_peer_addresses(peers, peer_id, addresses);
```

Comment:
Idk if we want to remove this. I'm not sure where we're gonna get the addresses though.

Reply:
I remembered that this is directly related to [this discussion](https://github.com/libp2p/rust-libp2p/discussions/4674) I had a couple of months ago.

Reply:
I  think I fixed it with some refactoring of `PeerReport` to move `Identify` and `Heartbeat` to `FuelBehavior` so we'd have access to the events, and with them the addresses, at the level we do this stuff ^^^.

It also cleans up the code a lot.

The UTs are passing again, so it should be good, but we can roll it back if needed.